python/src/specifications/specifications_impl.py

```diff
@@ -21,83 +33,131 @@ def spec_csv(architecture_file: str, output_directory: str) -> PluginExecutionRe
-                for req in section["requirements"]:
-                    ret_val.append(_gen_spec_line_from_req_dict(spec_name, section_name, req))
+        spec(str): A key for req_specs, pointing to the spec for which the current csv file is being generated
+    """
+    ret_val = []
+    if req_specs[spec].get_root_key() == "req_spec":  # make sure we're actually working with a spec here
+        if "requirements" in req_specs[spec].content:
+            for req in req_specs[spec].structure["req_spec"]["requirements"]:
+                if req in reqs:
+                    requirement = reqs[req]
+                    ret_val.append(_gen_spec_line_from_req_dict(req_specs[spec].name, "", requirement))
+
+        if "sections" in req_specs[spec].content:
+            for section in req_specs[spec].structure["req_spec"]["sections"]:
+                for req in req_specs[section].structure["req_spec"]["requirements"]:
+                    if req in reqs:
+                        requirement = reqs[req]
+                        ret_val.append(_gen_spec_line_from_req_dict(req_specs[spec].name, section, requirement))
     return ret_val
 
 
-def _gen_spec_line_from_req_dict(spec_name: str, section_name: str, req: dict) -> dict:
+def _gen_spec_line_from_req_dict(spec_name: str, section_name: str, req) -> dict:
+    """
+    Creates a line for a CSV file
+    Args:
+        spec_name(str): The name of the specification
+
+        section_name(str): The name of the section
+
+        req(Definition): The definition object of a requirement
+    """
     line = {}
     line["Spec Name"] = spec_name
     line["Section"] = section_name
-    line["ID"] = req["id"]
-    line["Requirement"] = req["shall"]
+    line["ID"] = req.structure["req"]["id"]
+    line["Requirement"] = req.structure["req"]["shall"]
     parent_ids = ""
-    if "parent" in req.keys():
-        for parent_id in req["parent"]["ids"]:
+    if "parents" in req.content:
+        for parent_id in req.structure["req"]["parents"]:
             if len(parent_ids) == 0:
                 parent_ids = f"{parent_id}"
             else:
                 parent_ids = f"{parent_ids} {parent_id}"
-        line["Parents"] = parent_ids
+
+    line["Parents"] = parent_ids
     child_ids = ""
-    if "child" in req.keys():
-        for child_id in req["child"]["ids"]:
+    if "children" in req.content:
+        for child_id in req.structure["req"]["children"]:
             if len(child_ids) == 0:
                 child_ids = f"{child_id}"
             else:
                 child_ids = f"{child_ids} {child_id}"
     line["Children"] = child_ids
     return line
+
+
+def _parent_child_check(rows, reqs) -> (bool, List[ExecutionMessage]):
+    """
+    Checks to confirm parents and children of requirements actually exist
+    Args:
+        rows(dict): A dictionary containing all the rows generated so far
+
+        reqs(dict): A dictionary containing all of the requirement definitions
```

Comment:
Missing your return, blank line following initial description, and there shouldn't be blank lines between the argument entries.

https://google.github.io/styleguide/pyguide.html#383-functions-and-methods

Reply:
I added the returns to the docstrings, and fixed the line spacing.